packages/webpack/src/migrations/update-15-6-3/webpack-config-setup.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import {
+  addProjectConfiguration,
+  readProjectConfiguration,
+  Tree,
+} from '@nrwl/devkit';
+import webpackConfigSetup from './webpack-config-setup';
+
+describe('15.6.3 migration (setup webpack.config file)', () => {
+  let tree: Tree;
+
+  beforeEach(async () => {
+    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
+
+    addProjectConfiguration(tree, 'app1', {
+      root: 'apps/app1',
+      targets: {
+        build: {
+          executor: '@nrwl/webpack:webpack',
+          options: {},
+        },
+      },
+    });
+    addProjectConfiguration(tree, 'app2', {
+      root: 'apps/app2',
+      targets: {
+        custom: {
+          executor: '@nrwl/webpack:webpack',
+          options: {},
+        },
+      },
+    });
+
+    addProjectConfiguration(tree, 'app3', {
+      root: 'apps/app3',
+      targets: {
+        custom: {
+          executor: '@nrwl/webpack:webpack',
+          options: {
+            webpackConfig: 'apps/app3/webpack.config.js',
+          },
+        },
+      },
+    });
+
+    addProjectConfiguration(tree, 'app4', {
+      root: 'apps/app4',
+      targets: {
+        custom: {
+          executor: '@nrwl/webpack:webpack',
+          options: {
+            webpackConfig: 'some/random/path/webpack.something.ts',
+          },
+        },
+      },
+    });
+
+    addProjectConfiguration(tree, 'app5', {
+      root: 'apps/app5',
+      targets: {
+        custom: {
+          executor: '@nrwl/webpack:webpack',
+          options: {
+            isolatedConfig: true,
+          },
+        },
+      },
+    });
+
+    await webpackConfigSetup(tree);
+  });
+
+  it('should create webpack.config.js file if it does not exist', () => {
+    expect(tree.read('apps/app1/webpack.config.js', 'utf-8')).toMatchSnapshot();
```

Comment:
Nitpick: There is no difference between app1 and app2 right? I think we only need to test one? Or you are checking that the migration migrates multiple projects? In that case maybe the test should be `should create webpack.config.js for projects that do not have one`.